students/eric_grandeo/lesson09/mailroom_oo.py

```diff
@@ -0,0 +1,199 @@
+
+
+    @property
+    def donor(self):
+        '''Return a dict of donor and donations'''
+        return {self.name: self.donations}
+
+
+    def add_donation(self, donation=[]):
+        """
+        Add donation to donor's history
+
+        Args:
+            donation (list):
+                Donation amount (must be a list)
+        """
+        self.donations.extend(donation)  
+
+
+    @property    
+    def sum_donations(self):
+        '''Return total donations made by donor'''
+        self._sum_donations = sum(self.donations)
+        return self._sum_donations
+
+    @property
+    def num_donations(self):
+        '''Return the number of donations made by donor'''
+        self._num_donations = len(self.donations)
+        return self._num_donations
+
+    @property
+    def avg_donations(self):
+        '''Return the average of donations made by donor'''
+        self._avg_donations = self.sum_donations/self.num_donations
+        return self._avg_donations
+
+    @property
+    def thank_you_letter(self):
+        '''Generate text for email thanking donor for their last donation'''
+        email_dict = dict(name=self.name, donation=int(self.donations[-1]))
+        result = """
+        Dear {name},
+        Thank you very much for the generous donation of ${donation:,.2f}
+        It is very much appreciated.
+        Respectfully,
+
+        Eric G.
+        """.format(**email_dict)
+
+        return result
+    
+    
+class DonorCollection:
+    """
+    Class representing a collection of Donor instances
+
+    Args:
+        donors (list):
+            List of donor objects (object)
+
+    Properties:
+        donor_names():
+            List of donor names. 
+
+        donor_dict():
+            A dictionary of donor name as key and donations as value
+
+        report_data():
+            A dictionary of all donors and donations, used to populate report
+
+    Methods:
+        add_donor(name, amount):
+            Either updates existing donor with amount, or Creates a new donor 
+            with amount if not in collection
+
+        donor_data(name):
+            Returns dictionary of donor instance 
+
+        donor_obj(name):
+            Returns the donor instance as an object
+
+        generate_report(report_data):
+            Generates report string from the data in the report_data property
+
+    """
+    def __init__(self):
+        self.donor_list = []
+
+    def add_donor(self, name, amount):
+        '''Either updates existing donor with amount, or Creates a new donor 
+            with amount if not in collection
+        '''
+        for donors in self.donor_list:
+            if donors.name == name:
+                donors.add_donation(amount)
+                return
+        
+        new_donor = Donor(name, amount)
+        self.donor_list.append(new_donor)
+
+    @property
+    def donor_names(self):
+        '''Returns a list of donor names'''
+        donor_name = []
```

Comment:
make sure that your data structures (which are plural) are represented by its name, so in this case you're holding `donor_names`